### PR TITLE
Zul'Aman raid improvements

### PR DESCRIPTION
**Pull request by grimgravy**
**Zul'Aman raid improvements**

## CHANGES PROPOSED:
* Fix: Spell Remove Amani Curse 43732 should only be castable on Forest Frog.
* Fix: Buffs resulting from usage of Amani Charm items only apply while with in Zul'Aman instance.
* Fix: Strange Gong position.
* Added: objecto Money Bag approximately 11 - 19 gold.
* Added: implementation of texts for the npcs.

From the patch cataclysm had some changes:
Mannuth replaced by [Mawago](https://wow.gamepedia.com/Mawago)
Deez replaced by [Kaldric](https://wow.gamepedia.com/Kaldrick)
Galathryn replaced by [Melasong](https://wow.gamepedia.com/Melasong)
Adarrah replaced by [Melissa](https://wow.gamepedia.com/Melissa)
Darwen replaced by [Arinoth](https://wow.gamepedia.com/Arinoth)
Fudgerick replaced by [Lenzo](https://wow.gamepedia.com/Lenzo)
Gunter replaced by [Harald](https://wow.gamepedia.com/Harald)
Kyren replaced by [Eulinda](https://wow.gamepedia.com/Eulinda)
Mitzi replaced by [Rosa](https://wow.gamepedia.com/Rosa)
Christian replaced by [Micah](https://wow.gamepedia.com/Micah)
Brennan replaced by [Tyllan](https://wow.gamepedia.com/)
Hollee replaced by [Relissa](https://wow.gamepedia.com/Relissa)

## ISSUES ADDRESSED:
Remove Amani Curse using on any target.
The npcs don't talk when using Remove Amani Curse.

## TESTS PERFORMED:
tested in windows 7 64bits

## HOW TO TEST THE CHANGES:
 Use Zul'Aman's raid teleport. There search for Forest Frog creatures using Remove Amani Curse.

## Target branch(es):
-  Dev

### CODE REVIEW

**Comment by grimgravy** [comment]
@Stoabrogga, what is the difference between "creature_text" and "broadcast_text"?

**Comment by Stoabrogga** [comment]
http://www.azerothcore.org/wiki/creature_text#broadcasttextid
http://www.azerothcore.org/wiki/broadcast_text
http://www.azerothcore.org/wiki/broadcast_text_locale

The broadcast texts are used in favor of the creature text if available:
https://github.com/azerothcore/azerothcore-wotlk/blob/4a8f1de5381d1cf53cdcc4c5bb7d9ef7c99a8a8f/src/server/game/Texts/CreatureTextMgr.cpp#L484-L488

File: data/sql/updates/pending_db_world/rev_1581083846978148700.sql [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+DELETE FROM `creature_equip_template` WHERE (`CreatureID` = 24406);
+INSERT INTO `creature_equip_template` (`CreatureID`, `ID`, `ItemID1`, `ItemID2`, `ItemID3`, `VerifiedBuild`) VALUES
+(24406, 1, 0, 12745, 0, 0);
+
+DELETE FROM `gameobject` WHERE (`id` = 187359);
+INSERT INTO `gameobject` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`, `spawntimesecs`, `animprogress`, `state`, `ScriptName`, `VerifiedBuild`) VALUES
+(20584, 187359, 568, 0, 0, 1, 1, 134.009, 1642.8, 42.0841, 3.14159, 0, 0, -1, 0, 25, 255, 1, '', 0);
+
+-- add restriction for Amani Charm buffs
+DELETE FROM spell_area WHERE area=3805 AND spell IN (43822,43816,43820,43818);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1581083846978148700.sql]: Missing `` for table name and columns.

File: data/sql/updates/pending_db_world/rev_1581083846978148700.sql [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+INSERT INTO spell_area (`spell`, `area`, `quest_start`, `quest_end`, `aura_spell`, `racemask`, `gender`, `autocast`, `quest_start_status`, `quest_end_status`) VALUES
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1581083846978148700.sql]: Here too for table name

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp [hunk 3]
```diff
@@ -52,56 +128,225 @@ class npc_forest_frog : public CreatureScript
...
+                                    DoCast(me, SPELL_STEALTH_, true);
+                                    break;
+                            }
+                            if (me->GetPositionY() > 1290.0f)
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 118.2742f, 1400.657f, -9.118711f);
+                            else
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 114.3155f, 1244.244f, -20.97606f);
+                            eventTimer = 0;
+                            break;
+						default:
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp]: Tabs here

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp [hunk 4]
```diff
@@ -52,56 +128,225 @@ class npc_forest_frog : public CreatureScript
...
+
                     if (roll_chance_i(2))
+                    {
                         DoCast(caster, SPELL_PUSH_MOJO, true);
+                        me->GetMotionMaster()->MovePoint(POINT_DESPAWN, caster->GetPositionX(), caster->GetPositionY(), caster->GetPositionZ());
+                    }
                     else
                         DoSpawnRandom();
                 }
+
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp]: Remove this line please.

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp [hunk 5]
```diff
@@ -52,56 +128,225 @@ class npc_forest_frog : public CreatureScript
...
+                            if (me->GetPositionY() > 1290.0f)
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 118.2742f, 1400.657f, -9.118711f);
+                            else
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 114.3155f, 1244.244f, -20.97606f);
+                            eventTimer = 0;
+                            break;
+						default:
+                            break;	
+                    }
+
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp]: Extra line here.

**Comment by grimgravy** [comment]
@Stoabrogga, I didn't know it was already in the DB, but it's not running on the server.

**Comment by Stoabrogga** [comment]
Yeah, I don't know where those texts came from but I always check the texts in the broadcast_text table when creating new creature text entries.

File: data/sql/updates/pending_db_world/rev_1581083846978148700.sql [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+(@ENTRY, 3, 0, 'You may want to consider leaving this place. I think being transformed into a frog is one of the better things that can happen to one here.', 12, 0, 100, 1, 0, 0, 23424, 0, 'SAY_BRENNAN_3');
+
+-- Hollee https://wow.gamepedia.com/Relissa
+SET @ENTRY := 24455;
+DELETE FROM `creature_text` WHERE `CreatureID`=@ENTRY;
+INSERT INTO `creature_text` (`CreatureID`,`GroupID`,`ID`,`Text`,`Type`,`Language`,`Probability`,`Emote`,`Duration`,`Sound`,`BroadcastTextId`, `TextRange`, `comment`) VALUES
+(@ENTRY, 0, 0, 'Oh! Back to my beautiful self? How lovely! I thought I would be a frog forever.', 12, 0, 100, 1, 0, 0, 23425, 0, 'SAY_HOLLEE_0'),
+(@ENTRY, 1, 0, 'It was just terrible. I was all slimy, and I kept licking my eyes!', 12, 0, 100, 1, 0, 0, 23426, 0, 'SAY_HOLLEE_1'),
+(@ENTRY, 2, 0, 'Thank goodness you came along when you did. Please, take one of these troll boxes to remember me by.', 12, 0, 100, 1, 0, 0, 23427, 0, 'SAY_HOLLEE_2'),
+(@ENTRY, 3, 0, 'I never want to see this nasty pond again. Thank you, $n. I\'ll never forget you!', 12, 0, 100, 1, 0, 0, 23428, 0, 'SAY_HOLLEE_3');
```
Comment (Stoabrogga) [inline data/sql/updates/pending_db_world/rev_1581083846978148700.sql]: Please add a newline to the end of the SQL script.

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp (hunk 3 above)
Comment (Stoabrogga) [reply src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp]: Please replace the tabs before "default:" with spaces.

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp [hunk 7]
```diff
@@ -52,56 +128,223 @@ class npc_forest_frog : public CreatureScript
...
+                                    break;
+                            }
+                            if (me->GetPositionY() > 1290.0f)
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 118.2742f, 1400.657f, -9.118711f);
+                            else
+                                me->GetMotionMaster()->MovePoint(POINT_DESPAWN, 114.3155f, 1244.244f, -20.97606f);
+                            eventTimer = 0;
+                            break;
+						default:
+                            break;
```
Comment (Stoabrogga) [inline src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp]: There's a tab after "break;". Please remove it.

**Comment by Stoabrogga** [comment]
Thanks @grimgravy for the updates, looks good now! Ready to be tested.

**Comment by Knindzagxg** [comment]
Looks good, no issues while testing.

**Comment by FrancescoBorzi** [comment]
good job @grimgravy and @Knindzagxg

**Pull request merged**